core/src/main/java/com/amplifyframework/auth/AuthSignInState.java

```diff
@@ -0,0 +1,114 @@
+ */
+
+package com.amplifyframework.auth;
+
+import androidx.annotation.NonNull;
+
+import java.util.Objects;
+
+// TODO: These values are currently coming from the ChallengeName values in AWS Cognito InitiateAuth call.
+//       The team decided to start with these since they are not Cognito specific but consider whether they
+//       should be renamed / supplemented. As future plugins need other states, they would be added here as well.
+public enum AuthSignInState {
+    /**
+     * Next challenge is to supply an SMS_MFA_CODE, delivered via SMS.
+     */
+    SMS_MFA("SMS_MFA"),
+
+    /**
+     *  Next challenge is to supply a SOFTWARE_TOKEN_MFA, delivered via software token.
+     */
+    SOFTWARE_TOKEN_MFA("SOFTWARE_TOKEN_MFA"),
+
+    /**
+     * If MFA is required, users who do not have at least one of the MFA methods set up are presented with this
+     * challenge. The user must set up at least one MFA type to continue to authenticate.
+     */
+    MFA_SETUP("MFA_SETUP"),
+
+    /**
+     * Must select an MFA type. Valid MFA options are SMS_MFA for text SMS MFA, and SOFTWARE_TOKEN_MFA for TOTP
+     * software token MFA.
+     */
+    SELECT_MFA_TYPE("SELECT_MFA_TYPE"),
+
+    /**
+     * Next challenge is to supply PASSWORD_CLAIM_SIGNATURE, PASSWORD_CLAIM_SECRET_BLOCK, and TIMESTAMP
+     * after the client-side SRP calculations.
+     */
+    PASSWORD_VERIFIER("PASSWORD_VERIFIER"),
+
+    /**
+     * This is returned if your custom authentication flow determines that the user should pass another challenge
+     * before tokens are issued.
+     */
+    CUSTOM_CHALLENGE("CUSTOM_CHALLENGE"),
+
+    /**
+     * If device tracking was enabled on your user pool and the previous challenges were passed, this challenge is
+     * returned so that Amazon Cognito can start tracking this device.
+     */
+    DEVICE_SRP_AUTH("DEVICE_SRP_AUTH"),
+
+    /**
+     * Similar to PASSWORD_VERIFIER, but for devices only.
+     */
+    DEVICE_PASSWORD_VERIFIER("DEVICE_PASSWORD_VERIFIER"),
+
+    /**
+     * This is returned if you need to authenticate with USERNAME and PASSWORD directly.
+     * An app client must be enabled to use this flow.
+     */
+    ADMIN_NO_SRP_AUTH("ADMIN_NO_SRP_AUTH"),
+
+    /**
+     * For users which are required to change their passwords after successful first login.
+     * This challenge should be passed with NEW_PASSWORD and any other required attributes.
+     */
+    NEW_PASSWORD_REQUIRED("NEW_PASSWORD_REQUIRED"),
+
+    /**
+     * The flow is completed and no further steps are needed.
+     */
+    DONE("DONE"),
+
+    /**
+     * Unknown sign-in state, potentially unsupported state.
+     */
+    UNKNOWN("UNKNOWN");
+
+    private final String value;
+
+    AuthSignInState(@NonNull String value) {
+        this.value = Objects.requireNonNull(value);
+    }
+
+    @NonNull
+    public String getValue() {
+        return value;
+    }
+
+    @NonNull
+    public static AuthSignInState getEnum(String value) {
```

Comment:
Naming for this method, I like any of these better:
1. `fromValue(String possibleValue)` (to match `getValue()`, which returns one.
2. `fromString(String possibleValue)`
3. `lookup(String possibleValue)`;
4. `from(String possibleValue)`

(`getEnum` is [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), technically - and [folks say not to use that](https://stackoverflow.com/a/111972/12442906), these days.)

Reply:
I see we're using fromString elsewhere so I'll update to that